2003/mtjin/AndroidArchitectureStudy/app/src/main/java/com/mtjin/androidarchitecturestudy/data/source/MovieRepositoryImpl.kt

```diff
@@ -0,0 +1,56 @@
+package com.mtjin.androidarchitecturestudy.data.source
+
+import com.mtjin.androidarchitecturestudy.data.Movie
+import com.mtjin.androidarchitecturestudy.data.source.local.MovieLocalDataSource
+import com.mtjin.androidarchitecturestudy.data.source.remote.MovieRemoteDataSource
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+class MovieRepositoryImpl(
+    private val movieRemoteDataSource: MovieRemoteDataSource,
+    private val movieLocalDataSource: MovieLocalDataSource
+) : MovieRepository {
+
+    private var movieList = ArrayList<Movie>()
+    override fun getSearchMovies(
+        query: String,
+        success: (List<Movie>) -> Unit,
+        fail: (Throwable) -> Unit
+    ) {
+        //TODO: 추후 네트워크 상태로 분기처리하게 변경
+        movieList.clear()
+        // 1. local 에서 검색
+        runBlocking {
+            launch {
+                movieList = movieLocalDataSource.getSearchMovies(query) as ArrayList<Movie>
```

Comment:
21번줄의 movieList와 25번줄에서 할당하는 movieList는 서로 다른 객체에요. 
clear()해주는 의미가 없어요.
```suggestion
        // 1. local 에서 검색
        runBlocking {
            launch {
                movieList.clear()
                movieList.addAll(movieLocalDataSource.getSearchMovies(query))
```

Reply:
e65107b
f3f2647
수정했습니다. 밑 코멘트도 동일합니다.